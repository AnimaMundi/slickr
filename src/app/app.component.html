<header><app-header></app-header></header>

<main>
  <app-search-form-container></app-search-form-container>

  <router-outlet></router-outlet>
</main>

<footer><app-footer></app-footer></footer>
