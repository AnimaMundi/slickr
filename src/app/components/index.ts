export * from './footer';
export * from './header';
export * from './search-form';
export * from './search-form-container';
