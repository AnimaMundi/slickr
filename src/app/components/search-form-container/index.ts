export * from './search-form-container.component';
