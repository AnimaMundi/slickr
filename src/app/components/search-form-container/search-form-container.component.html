<app-search-form
  [searchQuery]="searchQuery$ | async"
  (searchQueryChange)="onSearchQueryChanged($event)"
></app-search-form>
