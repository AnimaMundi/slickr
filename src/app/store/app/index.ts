export * from './app.reducer';
export * from './app.state';
export * from './router-state-serializer.class';
