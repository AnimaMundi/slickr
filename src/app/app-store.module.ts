import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { StoreRouterConnectingModule } from '@ngrx/router-store';

import { appReducer, metaReducers, CustomSerializer } from '@store/app';
import { SearchEffects } from '@store/search';
import { environment } from '@env';

@NgModule({
  imports: [
    StoreModule.forRoot(appReducer, { metaReducers }),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    EffectsModule.forRoot([SearchEffects]),
    StoreRouterConnectingModule.forRoot({
      serializer: CustomSerializer,
    }),
  ],
})
export class AppStoreModule {}
