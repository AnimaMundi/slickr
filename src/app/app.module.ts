import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';

import {
  MatToolbarModule,
  MatFormFieldModule,
  MatInputModule,
  MatIconModule,
} from '@angular/material';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { StoreRouterConnectingModule } from '@ngrx/router-store';

import { appReducer, metaReducers, CustomSerializer } from './store/app';
import { AppRoutingModule } from './app-routing.module';
import { environment } from '../environments/environment';

import { AppComponent } from './app.component';
import {
  HeaderComponent,
  FooterComponent,
  SearchFormComponent,
  SearchFormContainerComponent,
} from './components';

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
    SearchFormComponent,
    SearchFormContainerComponent,
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    ReactiveFormsModule,
    FormsModule,

    MatToolbarModule,
    MatFormFieldModule,
    MatInputModule,
    MatIconModule,

    AppRoutingModule,
    StoreModule.forRoot(appReducer, { metaReducers }),
    !environment.production ? StoreDevtoolsModule.instrument() : [],
    EffectsModule.forRoot([]),
    StoreRouterConnectingModule.forRoot({
      serializer: CustomSerializer,
    }),
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
