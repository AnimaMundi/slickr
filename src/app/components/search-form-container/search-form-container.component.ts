import { Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs';
import { Store, select } from '@ngrx/store';

import { AppState } from '@store/app';
import {
  getSearchQuery,
  SearchQueryChangedAction,
  SearchQueryFormSubmittedAction,
} from '@store/search';

@Component({
  selector: 'app-search-form-container',
  templateUrl: './search-form-container.component.html',
  styleUrls: ['./search-form-container.component.scss'],
})
export class SearchFormContainerComponent implements OnInit {
  public searchQuery$: Observable<string>;

  constructor(private readonly store$: Store<AppState>) {}

  public ngOnInit() {
    this.searchQuery$ = this.store$.pipe(select(getSearchQuery));
  }

  public onSearchQueryChanged(searchQuery: string): void {
    this.store$.dispatch(new SearchQueryChangedAction(searchQuery));
  }

  public onSearchQueryFormSubmitted(): void {
    this.store$.dispatch(new SearchQueryFormSubmittedAction());
  }
}
