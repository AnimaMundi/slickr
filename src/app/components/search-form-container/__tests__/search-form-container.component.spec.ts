import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { StoreModule } from '@ngrx/store';

import { appReducer } from '@store/app';

import { SearchFormContainerComponent } from '../search-form-container.component';

describe('SearchFormContainerComponent', () => {
  let component: SearchFormContainerComponent;
  let fixture: ComponentFixture<SearchFormContainerComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [StoreModule.forRoot(appReducer)],
      declarations: [SearchFormContainerComponent],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SearchFormContainerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should compile the template', () => {
    expect(fixture).toMatchSnapshot();
  });
});
