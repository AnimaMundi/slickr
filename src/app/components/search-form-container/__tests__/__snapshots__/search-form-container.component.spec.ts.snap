// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFormContainerComponent should compile the template 1`] = `
<app-search-form-container
  searchQuery$={[Function Store]}
  store$={[Function Store]}
>
  <app-search-form />
</app-search-form-container>
`;
